pkg/cluster/spec.go

```diff
@@ -142,22 +137,41 @@ func WithManagementCluster(cluster *types.Cluster) SpecOpt {
 	}
 }
 
-func NewSpec(clusterConfigPath string, cliVersion version.Info, opts ...SpecOpt) (*Spec, error) {
-	clusterConfig, err := eksav1alpha1.GetClusterConfig(clusterConfigPath)
-	if err != nil {
-		return nil, err
+func WithUserAgent(userAgent string) SpecOpt {
+	return func(s *Spec) {
+		s.userAgent = userAgent
 	}
+}
 
+func NewSpec(opts ...SpecOpt) *Spec {
 	s := &Spec{
 		releasesManifestURL: releasesManifestURL,
 		configFS:            configFS,
-		httpClient:          &http.Client{},
-		userAgent:           userAgent("cli", cliVersion.GitVersion),
+		userAgent:           userAgent("unknown", "unknown"),
```

Comment:
Do we want the default as this? I realize this is just code being moved, but saying something like "eks-a", "n/a" might be more descriptive if this function is being invoked.

Reply:
the `userAgent` method already adds the `eks-a` prefix
The first method argument is just the sub component name `eks-a-subcomponent`